indigo/views.py

```diff
@@ -982,25 +985,29 @@ def admin_projects_new(request):
         if form.is_valid():
             # process the data in form.cleaned_data as required
             # Save the event
-            id = form.cleaned_data["id"]
-            existing_record = Record.objects.filter(type=type, public_id=id)
-            if existing_record:
-                form.add_error("id", "This ID already exists")
-            else:
-                data = NewEventData(
-                    type,
-                    id,
-                    {"name": {"value": form.cleaned_data["name"]}},
-                    approved=True,
-                )
-                newEvent(
-                    [data], user=request.user, comment=form.cleaned_data["comment"]
+            last_record = (
+                Record.objects.filter(type=type).order_by("-public_id").first()
+            )
+            if last_record:
+                print(last_record.public_id)
+                m = re.search(
+                    "^INDIGO-POJ-([0-9][0-9][0-9][0-9])$", last_record.public_id
```

Comment:
Could also split by "-" and just take the last block, which would be more robust if any projects happen not to follow the "INDIGO-POJ-" pattern. But I just checked the data on all models and I think we are good there, so whatever you like.

Reply:
I deliberately picked the regex, because we'll find out sooner if something goes wrong (although conversely it does stop the creation of new projects until we fix it).